homeassistant/components/automation/numeric_state.py

```diff
@@ -38,16 +38,14 @@ def async_trigger(hass, config, action):
     time_delta = config.get(CONF_FOR)
     value_template = config.get(CONF_VALUE_TEMPLATE)
     async_remove_track_same = None
+    already_triggered = False
 
     if value_template is not None:
         value_template.hass = hass
 
     @callback
     def check_numeric_state(entity, from_s, to_s):
-        """Return True if they should trigger."""
-        if to_s is None:
```

Comment:
Don't remove this or we trigger it if we remove a state from state machine

Reply:
I removed this because the test seems to happen anyway in `condition.async_numeric_state`.

Reply:
You can believe me, it's needed. I add to my task list, that I will add a test for this case. But yeah it is rare.

Reply:
Pascal is correct. Needs to be added back.

Reply:
I trust you but I am really curious, can you explain why [this `async_numeric_state` check](https://github.com/home-assistant/home-assistant/blob/beb8c05d91276564406b60d520b3c191443559fa/homeassistant/helpers/condition.py#L149) is not sufficient?

Reply:
Why we should add a function frame to stack if we allready know the result of that function? On many hardware is our stack limited. We need also create a dict that will store into that frame at the end for eat only memory.

Safty first. The designe of core is that we check things on the right layer and make it robust.

Reply:
Safe memory and is optimize. We need not create a function frame to stack if we already know that we not need do this. The dict that follow eat in this case only memor and process in wors case a memory leak on heap

Reply:
Hmm, I am not sure that I agree with an optimization argument. This is optimizing the unlikely case while making the common case a little bit slower with two identical tests. Extra code also takes up memory :)

Never mind, I will add it back no problem. For sure, there are code style and robustness cases that can be made.

Thanks.

Reply:
Oh interesting, I wasn't aware there was another check in the condition. I still think it's good to just be explicit to people reading the trigger code what is going on . Thanks for putting it back 👍